### PR TITLE
♻️ deprecate JupyterDash in favor for updated Dash version

### PR DESCRIPTION
**Pull request by NielsPraet**
This PR starts the transition to using JupyterDash integrated in Dash itself (from Dash version 2.11.0 onwards). 
As the codebase for the Jupyter integration in Dash is currently pretty rigid, the `inline_persistent` mode can't be effectively implemented. For this reason, the JupyterDash dependency is still required when the `inline_persistent` mode is to be used. If this mode is not required, JupyterDash now is an optional dependency.

Might also introduce a fix for #123, but that remains to be seen.

### CODE REVIEW

File: examples/basic_example.ipynb [hunk 1]
```diff
@@ -735,7 +735,7 @@
     "The example below illustrates this behavior\n",
     "\n",
     "> **Note**: \n",
-    "> * you must have `kaleido` installed for this to work.\n",
+    "> * you must have `kaleido` and `jupyter-dash` installed for this to work.\n",
```
Comment (jonasvdd) [inline examples/basic_example.ipynb:738]: also `flask-cors`, right?

File: plotly_resampler/figure_resampler/figure_resampler.py [hunk 2]
```diff
@@ -31,174 +30,185 @@
 from .figure_resampler_interface import AbstractFigureAggregator
 from .utils import is_figure, is_fr
 
+try:
```
Comment (jonasvdd) [inline plotly_resampler/figure_resampler/figure_resampler.py:33]: Just a suggestion, maybe we can move the persistent inline output to a new file? (What are your opinions? @NielsPraet @jvdd)

File: plotly_resampler/figure_resampler/figure_resampler.py [hunk 3]
```diff
@@ -353,9 +363,11 @@ def __init__(
                     self.data[idx].update(graph_dict["data"][idx])
 
         # The FigureResampler needs a dash app
-        self._app: JupyterDash | dash.Dash | None = None
+        self._app: dash.Dash | None = None
         self._port: int | None = None
         self._host: str | None = None
+        # Certain functions will be different when using persistent inline (namely `show_dash` and `stop_callback`)
```
Comment (jonasvdd) [inline plotly_resampler/figure_resampler/figure_resampler.py:369]: doesn't this comment exceed the line-width of 88chars?

**Review by jonasvdd — APPROVED** [review]
LGTM, just a few comments!

File: plotly_resampler/figure_resampler/figure_resampler.py (hunk 2 above)
Comment (NielsPraet) [reply plotly_resampler/figure_resampler/figure_resampler.py:33]: Yes that would clean up the code just a bit, so I think this is a great idea

Comment (jonasvdd) [reply plotly_resampler/figure_resampler/figure_resampler.py:30]: 👍🏼

**Pull request merged**